test/unit/spreadElement.spec.ts

```diff
@@ -47,3 +47,23 @@ test("Spread Element Lua JIT", () => {
     const lua = util.transpileString(`[...[0, 1, 2]]`, options);
     expect(lua).toBe("local ____ = {unpack({\n    0,\n    1,\n    2,\n})}");
 });
+
+test("Spread Element Iterable", () => {
+    const code = `
+        const it = {
+            i: 10,
+            [Symbol.iterator]() {
+                return this;
+            },
+            next() {
+                --this.i;
```

Comment:
I think making this just count up is a bit clearer, first time I looked at this test I thought it was bugged and reversed the order.

Reply:
I usually do this on purpose to ensure we're not somehow getting array indicies instead of values.

Reply:
I get that, but I'd sooner opt for adding a constant to each value, or multiplying them by 3 or something like that.

Reply:
Ok, I'll do something different here